### PR TITLE
[ao][sparsity] Data Sparsity with Post Training Quantization

### PR DESCRIPTION
**Pull request by macandro96**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82759

Implementation of `post_training_sparse_quantize` that takes in a model
and applies sparsification and quantization to only `embeddings` & `embeddingbags`.
The quantization step can happen before or after sparsification depending on the `sparsify_first` argument.

Test Plan:
```python test/test_ao_sparsity.py TestQuantizationUtils```

### CODE REVIEW

File: test/ao/sparsity/test_data_sparsifier.py [hunk 1]
```diff
@@ -496,3 +498,98 @@ def test_nn_embeddings(self):
...
+        dequant_emb1 = torch.dequantize(model.emb1.weight())
+        dequant_embbag1 = torch.dequantize(model.embbag1.weight())
+
+        threshold = 1e-2
+
+        sl_emb1 = (torch.abs(dequant_emb1) < threshold).float().mean()
+        sl_embbag1 = (torch.abs(dequant_embbag1) < threshold).float().mean()
+
+        assert abs(sl_emb1 - 0.80) <= 0.05  # +- 5% leeway
+        assert abs(sl_embbag1 - 0.80) <= 0.05  # +- 5% leeway
```
Comment (z-a-f) [inline test/ao/sparsity/test_data_sparsifier.py:552]: You can use `assertAlmostEqual`: https://github.com/pytorch/pytorch/blob/7391dec96ab9f52b841973149922ff39ada489ff/test/ao/sparsity/test_data_sparsifier.py#L321

Comment (z-a-f) [reply test/ao/sparsity/test_data_sparsifier.py:552]: Is there a reason 5% is chosen?

**Review by z-a-f — APPROVED** [review]

File: test/ao/sparsity/test_data_sparsifier.py (hunk 1 above)
Comment (macandro96) [reply test/ao/sparsity/test_data_sparsifier.py:552]: Yes, I think some of the embeddings after quantization had sparsity levels of +- 5% from actual sparsity levels.

**Pull request closed without merging**